presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java

```diff
@@ -274,10 +276,22 @@ public void addColumn(long tableId, ColumnInfo column)
                 maxColumn(column.getColumnId()), columnType);
 
         try (Handle handle = dbi.open()) {
-            handle.execute(sql);
-        }
-        catch (DBIException e) {
-            throw metadataError(e);
+            int attempts = 0;
+            while (true) {
+                attempts++;
+                try {
+                    handle.execute(sql);
+                }
+                catch (DBIException e) {
+                    if (isSyntaxOrAccessError(e)) {
+                        // exit when column already exists
+                        return;
+                    }
+                    else if (attempts >= MAX_ADD_COLUMN_ATTEMPTS) {
```

Comment:
Why is it redundant? Without this check it'll never exit if an error is not a syntax one.